ambry-protocol/src/main/java/com/github/ambry/protocol/RequestDropper.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.network.NetworkRequest;
+import com.github.ambry.network.RequestResponseChannel;
+import com.github.ambry.server.EmptyRequest;
+import java.util.Collection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This class is used to query request channel for overflow requests and drop them. This is to make sure we don't let
+ * un-queued requests grow unbounded.
+ */
+public class RequestDropper implements Runnable {
+  private final RequestResponseChannel requestChannel;
+  private final RequestAPI requests;
+  private static final Logger logger = LoggerFactory.getLogger(RequestDropper.class);
+
+  public RequestDropper(RequestResponseChannel requestChannel, RequestAPI requests) {
+    this.requestChannel = requestChannel;
+    this.requests = requests;
+  }
+
+  public void run() {
```

Comment:
can this thread get blocked on sending a response ? if so, we might be blocked for a while.